src/util.ts

```diff
@@ -109,8 +109,9 @@ export async function fetchFrom(
   url: string,
   options?: http.RequestOptions
 ): Promise<string> {
+  const requestOptions: http.RequestOptions = { ...options, timeout: 5000 };
   const promise = new Promise<string>((resolve, reject) => {
-    http.get(url, options || {}, (resp) => {
+    http.get(url, requestOptions, (resp) => {
```

Comment:
This is not related to this PR, but we have to put `on("error", (e) => {  ... } )` outside of `http.get` like

```ts
    http
      .get(url, requestOptions, (resp) => {
        let body = "";
        resp.on("data", (chunk) => (body += chunk));
        resp.on("end", () => resolve(body));
      })
      .on("error", (e) => reject(e))
```

`resp.on("error", (e) => reject(e))` seems never reject on error.

Reply:
Also, it would be a good idea to show something on timeout 👍 

```ts
    http
      .get(url, requestOptions, (resp) => {
         // ...
      })
      .on("error", (e) => reject(e))
      .on("timeout", () => reject("Request timed out"));
```

Reply:
good points, I left `resp.on("error", (e) => reject(e));` though. I guess it feels more safe to react on this kind of event, although I'm not sure what will cause it to happen